core/protobuf/src/proto/dataserver/protocol.proto

```diff
@@ -75,6 +75,12 @@ message WriteRequest {
 
   optional bool eof = 6;
   optional bool cancel = 7;
+
+  // This is only applicable for ufs writes.
+  optional string ufs_path = 1000;
+  optional string owner = 1001;
+  optional string group = 1002;
+  optional int32 mode = 1003;
```

Comment:
I agree about the more efficient tag number though, I've changed it to use the next number. In the future we can logically group them and disregard the number.